src/messages/gateway.rs

```diff
@@ -208,4 +215,11 @@ pub enum ToGatewayMessage {
     LostWorker(LostWorkerMessage),
     WorkerStopped,
     GenericResourceNames(GenericResourceNames),
+    WorkerEventsResponse(WorkerEvents),
```

Comment:
Or would it be better to have WorkerEvents as an enum with different branches like WorkerConnectedEvent/TaskAssignedEvent, etc. and then have the data in them?

Reply:
We should definitely design a new message type that will be specific for monitoring, not reuse the existing messages.
So something like
```rust
enum MonitoringEventPayload {
   WorkerConnected(WorkerId)
}
```
etc.